test/network_autoconfiguration_test.rb

```diff
@@ -44,110 +44,136 @@ def probe_netcard_factory(num)
 end
 
 describe Yast::NetworkAutoconfiguration do
-  let(:instance) { Yast::NetworkAutoconfiguration.instance }
-  let(:network_interfaces) { double("NetworkInterfaces") }
-
-  before(:each) do
-    ifcfg_files = SectionKeyValue.new
-
-    # network configs
-    allow(Yast::SCR).to receive(:Dir) do |path|
-      case path.to_s
-      when ".network.section"
-        ifcfg_files.sections
-      when /^\.network\.value\."(eth\d+)"$/
-        ifcfg_files.keys(Regexp.last_match(1))
-      when ".modules.options", ".etc.install_inf"
-        []
-      else
-        raise "Unexpected Dir #{path}"
+  describe "it sets DHCLIENT_SET_DEFAULT_ROUTE properly" do
```

Comment:
np: You're describing `#configure_dhcp` and the description about what it does should be in an `it` block. I think the test need some refactoring but I guess it's out of the scope of this PR.